### DISCUSSION TITLE
We have any way to filter 404 logs in the production?

### DISCUSSION THREAD

**Discussion by meotimdihia** [fastify/fastify#5174] [Q&A]
I want to filter 404 (Error type: NotFoundError) errors in the output console.

![image](https://github.com/fastify/fastify/assets/300961/460d55ec-00a0-4360-b832-439678ce8bdb)

**Comment by metcoder95** [comment]
Hey!
The logs are `JSON` serialized and can be queried in the same way thanks to Pino. You will lose this capability if you set `pino-pretty` to post-process your logs. 

Depending on your setup, you can just deposit the logs somewhere that you can filter based on a predefined schema. e.g. send it to a backend that allows you to do `err.statusCode = 204`

**Reply by meotimdihia** [reply]
@metcoder95 do you have any suggestion about this problem?
What system should I use to see logs? I just want a simple setup :)
Currently, I am using Sentry to log errors ( filtered 404 errors - there is too much). 
But I want to filter 404 errors in the console in the production too.

**Reply by metcoder95** [reply]
If you have something local, you can use a simple file; you can do something like [`pino-file`](https://getpino.io/#/docs/transports?id=pinofile) should do the work; so you can use things like `jq` to query the file.

You can have multiple types of transport at pino, so you can maintain `pino-pretty` meanwhile writing to a file with `pino-file`.
You can have something as complex as an ELK stack, or use Grafana Loki, it really depends.
It is up to you what fits better for your problem 🙂 

Reference: https://getpino.io/#/docs/transports?id=v7-transports

**Comment by Uzlopak** [comment]
Do you want to ignore them from the logs?
Like your logs get spammed with 404 errors because some bot or so is going crazy?

**Reply by meotimdihia** [reply]
@Uzlopak I want to filter the logs at the console in the production server.

**Reply by Uzlopak** [reply]
The question is if you want to specifically want to disable only 404 errors or you want to generally disable?! Second thing you can do by setting the log level to "fatal" or "silent".

If you want to disable only 404 errors, than maybe it needs a more smarter approach.
